lib/perl/Genome/Model/Tools/Analysis/Coverage/CoveragePlot.pm

```diff
@@ -0,0 +1,1439 @@
+package Genome::Model::Tools::Analysis::Coverage::CoveragePlot;
+
+use strict;
+use Genome;
+use IO::File;
+use warnings;
+use Carp;
+
+use File::Basename;
+use File::Spec::Functions;
+use POSIX qw(ceil floor log log10);
+
+use Statistics::R;
+
+
+class Genome::Model::Tools::Analysis::Coverage::CoveragePlot{
+    is => 'Command',
+    has => [
+        roi_file_path => {
+	    is => 'String',
+	    is_optional => 0,
+            doc => 'Region-of-interest file in the BED format',
+        },
+	
+	output_directory => {
+	    is => 'String',
+	    is_optional => 0,
+	    doc => 'Directory to save output files such as PDF file and tables for plotting, and ref-cov stats files',
+	},
+	
+	bam_files => {
+	    is => 'String',
+	    is_optional => 1,
+	    doc => 'Comma-separated list of bam files to generate coverage plots from, Must specify either --bam_files or --refalign_models',
+	},
+	
+	refalign_models => {
+	    is => 'Number',
+	    is_optional => 1,
+	    doc => 'Comma-separated list of refalign model IDs to generate coverage plots from, bam_files will override refalign_models if given',
+	},
```

Comment:
You should make this a `is => 'Genome::Model::ReferenceAlignment'`. This will automatically get the model object for each id for you and it will do some error checking that the id is indeed a reference alignment model. You don't need to do the retrieval and check on lines 188 and 189. 

This should also be a `is_many => 1`